pyogrio/_io.pyx

```diff
@@ -1390,8 +1390,7 @@ def ogr_open_arrow(
         stream_ptr = <uintptr_t> &stream
 
         if skip_features:
-            # only supported for GDAL >= 3.8.0; have to do this after getting
-            # the Arrow stream
+            # For GDAL < 3.8.0 this needs to be set after getting the Arrow stream
```

Comment:
I don't think this change to the comment is needed?  If the goal is to clarify the previous comment, we could use something like:

```
# skip_features is only supported for GDAL >= 3.8.0; otherwise this is responsibility of the caller
# need to set OGR_L_SetNextByIndex after getting the Arrow stream
```

Reply:
If I recall correctly from when testing this, in GDAL < 3.8 there were 2 issues:

- the combination of a sql filter with a LIMIT clause + OGR_L_SetNextByIndex gave wrong results. This was fixed in gdal >= 3.8.
- OGR_L_SetNextByIndex had to be run after OGR_L_GetArrowStream because in OGR_L_GetArrowStream the index was reset to 0. This was removed in GDAL 3.8 so for gdal > 3.8 OGR_L_SetNextByIndex can be called before OGR_L_GetArrowStream and it will work.

Because `skip_features` is only enabled from GDAL >= 3.8 it all doesn't really matter... because in that version all bugs were fixed. So, I think the current comment isn't really accurate, but I suppose it is not really important.

Reply:
Anyway, I'll give it another try based on your new suggestion.